### ISSUE TITLE
Issue Running Hidden Service on Fedora

### ISSUE THREAD

**Issue by Rspigler** [bitcoin/bitcoin#13343]
Describe the issue:
Some recent update to my Fedora system killed the connections I had to my hidden service.  I've attempted re-installing, restarting tor (`systemctl restart tor.service`), and restarting Bitcoin Core, but to no avail.

I have the following in my `torrc` file:

> HiddenServiceDir /var/lib/tor/bitcoin-service/
>HiddenServicePort 8333 127.0.0.1:8333

And the following in bitcoin.conf:

> onion=127.0.0.1:9050
>listen=1
>externalip=ONION_ADDR
>discover=1

ONION_ADDR was found by `cat /var/lib/tor/bitcoin-service/hostname`

What behavior did you expect?:
Not losing my connections, or regaining connections upon reinstall.

How reliably can you reproduce the issue, what are the steps to do so?:
Just install Fedora 28, Bitcoin Core, and attempt to set up a Tor node.

What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)?:
Bitcoin 0.16.0 from BitcoinCore.org (verified!).

**Comment by marcoagner** [comment]
It may be an issue with your Fedora/SELinux policy.
Have you tried investigating these?

**Comment by Rspigler** [comment]
Sorry, been out of the country for the last two weeks.

It is not a SELinux policy issue - running `setenforce 0` and restarting Bitcoin Core still doesn't allow any connections to my hidden service.

I should note that updating to 0.16.1 has not changed this (not that there was any indication there was supposed to be a fix for this including in the update).

**Comment by Rspigler** [comment]
So, had been testing my connections using https://bitnodes.earn.com

This is still failing for some reason, but I have gained 2 connections back to my Tor node today.  This is after I attempted restarting `firewalld` (in addition to restarting Bitcoin Core, SELinux, and tor like mentioned above in my initial post).  Not sure if that's what fixed this, or the recent kernel update from 4.16.16 to 4.17.2 that Fedora pushed? But anyway, closing this now.
